src/signal.rs

```diff
@@ -521,6 +521,15 @@ pub struct GenMut<G, F> {
     frame: core::marker::PhantomData<F>,
 }
 
+/// A signal that calls its enclosing function and returns the original value. The signal may
+/// mutate state.
+#[derive(Clone)]
+pub struct Inspect<S, Func, F> {
+    signal: S,
+    func: Func,
+    frame: core::marker::PhantomData<F>,
```

Comment:
This `F` type parameter should not be necessary as the frame type can be accessed via `Signal`'s associated `Frame` type (e.g. `S::Frame`) in the `inspect` method and the `Signal` implementation for `Inspect`.

Reply:
The F type is the output frame, which might be a different number of channels than the input frame.